drivers: fix gpio_mcux_lpc_port_set_masked_raw

agansari opened the pull request:
Fixes gpio_mcux_lpc_port_set_masked_raw function inside gpio_mcux_lpc
driver. Writing masked data is correctly done using device registers.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

Fixes #29070

agansari committed:
drivers: fix gpio_mcux_lpc_port_set_masked_raw

Fixes gpio_mcux_lpc_port_set_masked_raw function inside gpio_mcux_lpc
driver. Writing masked data is correctly done using device registers.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

pabigot reviewed (approved):
Haven't dug into the datasheet but I can believe this is right.

The existing test is inadequate.  Would you be interested in adding a commit to fix that?  I think this code:
```
        rc = gpio_port_set_masked_raw(dev, BIT(PIN_IN), 0);
        zassert_equal(rc, 0,
                      "set_masked_raw low failed");
        zassert_equal(raw_in(), true,
                      "set_masked_raw low affected other pins");
```
inserted at https://github.com/zephyrproject-rtos/zephyr/blob/4d0750920b52e0ce8a3e188d08a1769ff7ec527b/tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c#L185 would do it: it should detect that the attempt to set only the input pin to zero also affected the output pin.

That should make the test fail on the unpatched code.

agansari commented:
@pabigot sure

> Would you be interested in adding a commit to fix that?

agansari committed:
tests: gpio: gpio_port_set_masked_raw overwrite check

Code checks if gpio_port_set_masked_raw overwrites masked pins.
It should detect that the attempt to set only the input pin to zero also
affected the output pin.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari commented:
@pabigot I've added the test and ran it with and without the previous commit, the result is as expected, PASS with fix, "set_masked_raw low affected other pins" i.e. FAIL without the mask fix.

Hope this PR does not transform in a fix all non-working pin masks.

pabigot commented on tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c:
```diff
@@ -183,6 +183,12 @@ static int bits_physical(void)
 
+	rc = gpio_port_set_masked_raw(dev, BIT(PIN_IN), 0);
+	zassert_equal(rc, 0,
+					"set_masked_raw low failed");
```
Indentation seems wrong.

pabigot commented:
> Hope this PR does not transform in a fix all non-working pin masks.

Thanks; if it does cause other platforms to break that's a good thing to know.

pabigot reviewed (changes requested):
Please fix indentation in test case.
